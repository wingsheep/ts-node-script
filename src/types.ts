export interface DataItem {
  identifier: string
  name: string
  [key: string]: unknown
}
