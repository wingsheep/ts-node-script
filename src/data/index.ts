export const template = `
<ElForm ref="conditionForm" :model="conditions" inline>
    <ElFormItem prop="orderCode">
      <BaseTooltip content="订单号" inner-tag="div">
        <SearchInput
          v-model="conditions.orderCode"
          type="orderCode"
          :order-type="0"
          :status="currentStatusValue"
        />
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem v-if="processModeListMap?.selfOrder" prop="selfOrder">
      <BaseTooltip content="业务单号" inner-tag="div">
        <SearchInput
          v-model="conditions.selfOrder"
          type="selfOrder"
          :status="currentStatusValue"
        />
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="fbaNumber">
      <BaseTooltip content="FBA编号" inner-tag="div">
        <ElInput
          v-model="conditions.fbaNumber"
          placeholder="请输入FBA编号"
        ></ElInput>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem
      v-if="isPacked || isShipping || isReceipted || isAll"
      prop="transportCode"
    >
      <BaseTooltip content="运输单号" inner-tag="div">
        <ElInput
          v-model="conditions.transportCode"
          placeholder="请输入运输单号"
          maxlength="50"
          clearable
        ></ElInput>
      </BaseTooltip>
    </ElFormItem>
    <!--<ElFormItem
            v-if="isPacked || isShipping || isReceipted"
            prop="containerCode"
        >
            <BaseTooltip content="托盘号" inner-tag="div">
                <ElInput
                    v-model="conditions.containerCode"
                    placeholder="请输入托盘号"
                    maxlength="10"
                    clearable
                ></ElInput>
            </BaseTooltip>
        </ElFormItem>-->

    <ElFormItem prop="customerBizKey">
      <BaseTooltip content="客户账号" inner-tag="div">
        <CustomerEditor v-model="conditions.customerBizKey"></CustomerEditor>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem prop="dstId">
      <BaseTooltip content="目的地" inner-tag="div">
        <ElSelect
          v-model="conditions.dstId"
          placeholder="请选择目的地"
          filterable
          clearable
        >
          <ElOption label="全部目的地" value=""></ElOption>
          <ElOption
            v-for="item in destinations"
            :key="item.id"
            :value="item.id"
            :label="item.name"
          >
            {{ item.name }}({{ item.code }})
          </ElOption>
        </ElSelect>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem prop="channelTypeId">
      <BaseTooltip content="运输方式" inner-tag="div">
        <ElSelect
          v-model="conditions.channelTypeId"
          placeholder="请选择运输方式"
          filterable
          clearable
        >
          <ElOption label="全部运输方式" value=""></ElOption>
          <ElOption
            v-for="item in channelTypes"
            :key="item.bizKey"
            :value="item.bizKey"
            :label="item.name"
          ></ElOption>
        </ElSelect>
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="planId">
      <BaseTooltip content="渠道" inner-tag="div">
        <ChaneelSeletor
          v-model="conditions.planId"
          placeholder="请选择渠道"
          clearable
          model-value-type="prop"
        ></ChaneelSeletor>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem
      v-if="isShipping || isReceipted || isAll"
      prop="transportStatusCondition"
    >
      <BaseTooltip content="头程运输状态" inner-tag="div">
        <ElSelect
          v-model="conditions.transportStatusCondition"
          placeholder="请选择头程运输状态"
          clearable
        >
          <ElOption label="全部头程运输状态" value=""></ElOption>
          <ElOption
            v-for="item in orderHeadTransportStatuses"
            :key="item.value"
            :label="item.name"
            :value="item.value"
          ></ElOption>
        </ElSelect>
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="receiverType">
      <BaseTooltip content="收件类型" inner-tag="div">
        <ElSelect
          v-model="conditions.receiverType"
          placeholder="请选择收件类型"
          clearable
          @change="handleChangeReceiverType"
          @clear="handleChangeReceiverType"
        >
          <ElOption
            v-for="opt in DestAddressType"
            :key="opt.value"
            :label="opt.label"
            :value="opt.value"
          />
        </ElSelect>
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem v-if="conditions.receiverType" prop="receiverKeyword">
      <BaseTooltip
        v-if="conditions.receiverType === -1"
        content="收件人姓名"
        inner-tag="div"
      >
        <ElInput
          v-model="conditions.receiverKeyword"
          placeholder="请输入收件人姓名"
          clearable
        />
      </BaseTooltip>
      <BaseTooltip
        v-else-if="conditions.receiverType === 1"
        content="海外仓(FBA)"
        inner-tag="div"
      >
        <DestGroupCascader
          v-model="conditions.destinationId"
          placeholder="请选择海外仓(FBA)"
          clearable
          filter="def-ban"
          type="1"
        />
      </BaseTooltip>
      <BaseTooltip
        v-else-if="conditions.receiverType === 2"
        content="自提点"
        inner-tag="div"
      >
        <DestGroupCascader
          v-model="conditions.destinationId"
          placeholder="请选择自提点"
          clearable
          type="2"
        />
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="warehouseId">
      <BaseTooltip content="所属仓库" inner-tag="div">
        <BelongWarehouseSelect
          v-model="conditions.warehouseId"
          filter="def-ban"
        />
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="orderBelonging">
      <BaseTooltip content="订单归属人员" inner-tag="div">
        <OrderBelongingSelector v-model="conditions.orderBelonging" />
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="businessTimeRange">
      <BaseTooltip content="业务日期" inner-tag="div">
        <ElDatePicker
          v-model="conditions.businessTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="业务日期 (始)"
          end-placeholder="业务日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem prop="createTimeRange">
      <BaseTooltip content="建单时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.createTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="建单日期 (始)"
          end-placeholder="建单日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem v-if="isPrePayment || isAll" prop="checkedTimeRange">
      <BaseTooltip content="核单时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.checkedTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="核单日期 (始)"
          end-placeholder="核单日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem v-if="isPreShipping || isAll" prop="paymentTimeRange"
      ><BaseTooltip content="放单时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.paymentTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="放单日期 (始)"
          end-placeholder="放单日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem v-if="isPacked || isAll" prop="packedTimeRange">
      <BaseTooltip content="装箱时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.packedTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="装箱日期 (始)"
          end-placeholder="装箱日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem
      v-if="isShipping || isReceipted || isAll"
      prop="shippingTimeRange"
    >
      <BaseTooltip content="发货时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.shippingTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="发货日期 (始)"
          end-placeholder="发货日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>
    <ElFormItem v-if="isReceipted || isAll" prop="confirmReceiptRange">
      <BaseTooltip content="签收时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.confirmReceiptRange"
          type="daterange"
          range-separator="-"
          start-placeholder="签收日期 (始)"
          end-placeholder="签收日期 (终)"
          format="YYYY-MM-DD"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem v-if="isCanceled || isAll" prop="cancelTimeRange">
      <BaseTooltip content="退件时间" inner-tag="div">
        <ElDatePicker
          v-model="conditions.cancelTimeRange"
          type="daterange"
          range-separator="-"
          start-placeholder="退件日期 (始)"
          end-placeholder="退件日期 (终)"
          format="YYYY-MM-DD"
          value-format="YYYY-MM-DD HH:mm:ss"
          :default-time="defaultTime"
          clearable
        >
        </ElDatePicker>
      </BaseTooltip>
    </ElFormItem>

    <ElFormItem>
      <BaseButton type="theme" :disabled="loading" @click="searchList">
        查询
      </BaseButton>
      <BaseButton type="plain" :disabled="loading" @click="resetQuery">
        重置
      </BaseButton>
    </ElFormItem>
  </ElForm>
`
