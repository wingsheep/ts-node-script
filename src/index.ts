import fs from 'fs/promises'
import path from 'path'
import type { DataItem } from './types'

/**
 * 生成枚举文件
 *
 * @default ./src/enumByGenerate.ts
 */
export const generateEnumFile = async (data: DataItem[], ...pathSegments: string[]): Promise<string> => {
  const _pathSegments = pathSegments && pathSegments.length ? pathSegments : ['./src/enumByGenerate.ts']
  let output = '// eslint-disable-next-line @typescript-eslint/comma-dangle\nexport enum AUTH_KEY {\n  '
  output += data.map(item => `${item.identifier} = '${item.name}'`).join(',\n  ')
  output += '\n}\n'
  await fs.writeFile(path.resolve(..._pathSegments), output)
  return output
}
