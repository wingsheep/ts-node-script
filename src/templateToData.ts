import console from 'console'
import fs from 'fs/promises'
import path from 'path'

interface DataItem {
  label: string
  prop: string
  slot?: string
  isShow?: string | boolean
  [key: string]: unknown
}

/**
 * 获取tag Dom
 * @param {string} tag
 * @param {string} str
 * @returns
 */
function getTagStr(tag: string, str: string, closeTag = true, inner = false) {
  let reg
  if (closeTag) {
    reg = new RegExp(`\\<${tag}[\\s\\S]*?<\\/${tag}>`, 'g')
    const tagStr = str.match(reg) || ''
    if (inner) {
      // 获取html标签内的内容
      const innerReg = /(?<=((<[a-zA-Z-]+?){0,1}>))([\s\S]+)(?=([\s]{0,1}<\/[a-zA-Z-]+(>{0,1})))/g
      const innerContent = tagStr && tagStr[0] && tagStr[0].match(innerReg)
      return innerContent && innerContent[0]
    }
    else {
      return tagStr
    }
  }
  else {
    reg = new RegExp(`\\<${tag}([\\s\\S]*?)>`, 'g')
    const matches = str.match(reg)
    return matches && matches[0]
  }
}

/**
 * 获取属性内容
 *
 * @param {string} attr
 * @param {string} str
 * @returns
 */
function getAttrStr(attr: string, str: string, inner = true) {
  const reg = new RegExp(`${attr}="([\\s\\S]*?)"`, 'g')
  const matches = str && str.match(reg)
  const content = matches && matches[0]
  const innerReg = new RegExp(`(?<=${attr}=").+(?=\\")`, 'g')
  const innerMatches = content && content.match(innerReg)
  return inner ? innerMatches && innerMatches[0] : matches
}

interface Options {
  /** 是否返回template */
  isReturnTemplate?: boolean
  /** 是否写死isShow */
  isShow?: boolean
}
/**
 * 模板转换为数据格式
 *
 */
export const templateToData = (str: string, options: Options = {
  isReturnTemplate: false,
  isShow: false,
}): any => {
  // 去除html注释
  str = str.replace(/<!--[\s\S]*?-->/g, '')
  // 匹配ElFormItem
  const elFormItems = getTagStr('ElFormItem', str, true) as RegExpMatchArray
  if (elFormItems === null) {
    console.error('没有找到ElFormItem')
    return []
  }
  else {
    const data: DataItem[] = []
    elFormItems.forEach((elFormItem: any) => {
      const elFormItemPrefixTagStr: any = getTagStr('ElFormItem', elFormItem, false)
      const baseTooltipPrefixTagStr: any = getTagStr('BaseTooltip', elFormItem, false)
      const baseTooltipTagStr: any = getTagStr('BaseTooltip', elFormItem, true, true)
      const prop = getAttrStr('prop', elFormItemPrefixTagStr, true) as string
      const isShow = getAttrStr('v-if', elFormItemPrefixTagStr, true) as string
      const label = getAttrStr('content', baseTooltipPrefixTagStr, true) as string
      if (label && prop) {
        data.push({
          label,
          prop,
          slot: prop,
          isShow: options.isShow ? true : isShow || true,
          template: baseTooltipTagStr,
        })
      }
    })
    if (options.isReturnTemplate) {
      return data
    }
    else {
      return data.map(({ label, prop, isShow }) => {
        return {
          label,
          prop,
          slot: prop,
          isShow,
        }
      })
    }
  }
}
/**
 * 生成组件模板
 * @param str
 * @returns
 */
export const generateTemplate = async (str: string, isShow = false, ...pathSegments: string[]): Promise<string> => {
  const _pathSegments = pathSegments && pathSegments.length ? pathSegments : ['./src/templateByGenerate.vue']
  const templateData = templateToData(str, { isReturnTemplate: true, isShow })
  let output = '<template><BaseConditionsForm :columns="columns">'
  templateData.forEach((item: DataItem) => {
    const itemStr = `<template #${item.slot}>
      ${item.template}
      </template>
    `
    output += itemStr
  })
  output += `</BaseConditionsForm></template><script lang="ts" setup>
  import { ref } from 'vue'
  const columns = ref([
 `
  templateData.forEach((item: DataItem) => {
    output += `{
      label: '${item.label}',
      prop: '${item.prop}',
      slot: '${item.slot}',
      isShow: ${item.isShow},
    },`
  })
  output += '])</script>'
  await fs.writeFile(path.resolve(..._pathSegments), output)
  return output
}
