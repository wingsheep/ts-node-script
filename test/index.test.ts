import { describe, expect, it } from 'vitest'
import { generateTemplate, templateToData } from '../src/templateToData'
import { template } from '../src/data/index'

describe('should', () => {
  // it('exported', () => {
  //   expect(templateToData(template)).toMatchInlineSnapshot()
  // })
  it('exported', () => {
    expect(generateTemplate(template)).toMatchInlineSnapshot('Promise {}')
  })
})
