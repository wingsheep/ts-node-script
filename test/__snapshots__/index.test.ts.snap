// Vitest Snapshot v1

exports[`should > exported 1`] = `
[
  {
    "businessOwnership": "kygj",
    "id": "1440551560446349314",
    "identifier": "RailTransport001",
    "name": "头程/铁路头程",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539814234195976193",
    "identifier": "BindOnlineAccount001",
    "name": "系统/业务员管理/绑定线上账号",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417408641469845505",
    "identifier": "FBScheme004",
    "name": "渠道",
  },
  {
    "businessOwnership": "kygj",
    "id": "1526143899416354818",
    "identifier": "FBAOrder00",
    "name": "FBA物流",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417400500455804929",
    "identifier": "superAminPrivilege01",
    "name": "超级管理员权限",
  },
  {
    "businessOwnership": "kygj",
    "id": "1526145255560667138",
    "identifier": "FBBDGoodMG001",
    "name": "设置/商品管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412661001392129",
    "identifier": "BusinessAuthority001",
    "name": "系统/权限管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539813821459685377",
    "identifier": "FBAOrder0017",
    "name": "FBA物流/FBA订单/已签收",
  },
  {
    "businessOwnership": "kygj",
    "id": "1440551400257490946",
    "identifier": "AirTransport001",
    "name": "头程/空运头程",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539817651597434882",
    "identifier": "DeleteBundle001",
    "name": "专线物流/集货管理/删除包裹",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417409249748783106",
    "identifier": "FBBDStores001",
    "name": "设置/收货仓库",
  },
  {
    "businessOwnership": "kygj",
    "id": "1537681106073509890",
    "identifier": "SurchargeSetting001",
    "name": "渠道/附加费设置",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417410795693740033",
    "identifier": "FBBDRechargeTemp01",
    "name": "设置/财务设置/充值项",
  },
  {
    "businessOwnership": "kygj",
    "id": "1451467424686800897",
    "identifier": "CustomerManag004",
    "name": "系统/业务员管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539812905071370241",
    "identifier": "FBAOrder0011",
    "name": "FBA物流/FBA订单/待入库",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417410130963664897",
    "identifier": "FBBDPaymentWay001",
    "name": "设置/财务设置/收款账户",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403540575424513",
    "identifier": "FBOrder0021",
    "name": "专线物流/专线订单/待核单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417411286599274497",
    "identifier": "FinancialRechRev001",
    "name": "收入/充值审核",
  },
  {
    "businessOwnership": "kygj",
    "id": "1452889950797299714",
    "identifier": "SchemeSupplier001",
    "name": "供应商",
  },
  {
    "businessOwnership": "kygj",
    "id": "1526143461442936834",
    "identifier": "LineOrder00",
    "name": "专线物流",
  },
  {
    "businessOwnership": "kygj",
    "id": "1504695654008455169",
    "identifier": "PaidService001",
    "name": "设置/增值服务",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539813334928809986",
    "identifier": "FBAOrder0013",
    "name": "FBA物流/FBA订单/已核单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1537680894382792705",
    "identifier": "Channel001",
    "name": "渠道/渠道管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539813483533000706",
    "identifier": "FBAOrder0014",
    "name": "FBA物流/FBA订单/待出运",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417410916128985089",
    "identifier": "FBBDOrderFlowSet001",
    "name": "设置/下单流程设置",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417410416088256514",
    "identifier": "FBBDGoodsType001",
    "name": "设置/货物类型",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403138048069633",
    "identifier": "FBPreOrder0012",
    "name": "专线物流/集货管理/已入库",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091594363113473",
    "identifier": "ReceivableManager001",
    "name": "收入",
  },
  {
    "businessOwnership": "kygj",
    "id": "1418121230515724288",
    "identifier": "FBBDDistSet001",
    "name": "设置/渠道商设置",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417402919734546434",
    "identifier": "FBPreOrder001",
    "name": "专线物流/集货管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091278834012162",
    "identifier": "PayableSettlement001",
    "name": "成本/应付报表",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091433561886722",
    "identifier": "ReceivableSettleR001",
    "name": "收入/结算记录",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091664202469378",
    "identifier": "PayableManager001",
    "name": "成本",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417410023216189442",
    "identifier": "FBBDSettlementWay001",
    "name": "设置/财务设置/结算方式",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412531812634625",
    "identifier": "BusinessManag001",
    "name": "系统",
  },
  {
    "businessOwnership": "kygj",
    "id": "1430416988710768642",
    "identifier": "FBHeadTransport001",
    "name": "头程",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539813667692306433",
    "identifier": "FBAOrder0016",
    "name": "FBA物流/FBA订单/已出运",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539868721992392706",
    "identifier": "FBOrder0028",
    "name": "专线物流/专线订单/全部",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417400994746142721",
    "identifier": "brandLogoUpdate01",
    "name": "超级管理员权限/品牌Logo修改",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417409139010768898",
    "identifier": "FBBDNotice001",
    "name": "设置/公告管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417401874434297858",
    "identifier": "SiteManagement002",
    "name": "客户端/移动端",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403941458612225",
    "identifier": "FBOrder0025",
    "name": "专线物流/专线订单/已出运",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091513949917186",
    "identifier": "PayableSettleR001",
    "name": "成本/结算记录",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403419246792706",
    "identifier": "FBOrder002",
    "name": "专线物流/专线订单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412840584712194",
    "identifier": "RoleMangement001",
    "name": "系统/角色管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1496780529364299778",
    "identifier": "BusinessManag003",
    "name": "系统/子公司管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539872077928148993",
    "identifier": "FBAOrder0015",
    "name": "FBA物流/FBA订单/已装箱",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403041621020673",
    "identifier": "FBPreOrder0011",
    "name": "专线物流/集货管理/待入库",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417401450566324225",
    "identifier": "companyUserRoleSet01",
    "name": "超级管理员权限/企业角色设置权限",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539817424257769474",
    "identifier": "DeleteFBAOrder001",
    "name": "FBA物流/FBA订单/删除FBA订单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417401644250894337",
    "identifier": "SiteManagement00",
    "name": "客户端",
  },
  {
    "businessOwnership": "kygj",
    "id": "1496780754367737858",
    "identifier": "FinancialReport001",
    "name": "报表/财务报表",
  },
  {
    "businessOwnership": "kygj",
    "id": "1514823201649549314",
    "identifier": "FinancialSetup001",
    "name": "设置/财务设置",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412227964669953",
    "identifier": "CustomerManag002",
    "name": "客户/客户管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417401135116914690",
    "identifier": "i18n",
    "name": "超级管理员权限/官网-外语页面开关",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417401776988033025",
    "identifier": "SiteManagement001",
    "name": "客户端/电脑端",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417401968592228353",
    "identifier": "SiteManagement003",
    "name": "客户端/小程序",
  },
  {
    "businessOwnership": "kygj",
    "id": "1522905321991266305",
    "identifier": "BaseDataCurrency001",
    "name": "设置/财务设置/汇率币别",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539868828435439617",
    "identifier": "FBPreOrder0016",
    "name": "专线物流/集货管理/无主件",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417409019309527042",
    "identifier": "FBBDBanner001",
    "name": "设置/轮播管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539817555468181506",
    "identifier": "DeleteFBOrder001",
    "name": "专线物流/专线订单/删除专线订单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417408920151986177",
    "identifier": "FBBaseData00",
    "name": "设置",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417408524222271489",
    "identifier": "FBTransport003",
    "name": "头程/海运头程",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417404026334547970",
    "identifier": "FBOrder0026",
    "name": "专线物流/专线订单/已退件",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412141566201858",
    "identifier": "CustomerManag001",
    "name": "客户",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403730338320385",
    "identifier": "FBOrder0023",
    "name": "专线物流/专线订单/待出运",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403299394555906",
    "identifier": "FBPreOrder0014",
    "name": "专线物流/集货管理/已取消",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403640311779329",
    "identifier": "FBOrder0022",
    "name": "专线物流/专线订单/已核单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1418121230522580992",
    "identifier": "FBCustomerGroup001",
    "name": "渠道/报价客户组",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417408801461571586",
    "identifier": "FBSchemeFreight001",
    "name": "渠道/报价管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417411382195851266",
    "identifier": "FinancialBalance001",
    "name": "收入/账户明细",
  },
  {
    "businessOwnership": "kygj",
    "id": "1496780643663278081",
    "identifier": "ReportManager001",
    "name": "报表",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091354532810753",
    "identifier": "ReceivableSettle001",
    "name": "收入/应收报表",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417409556998328322",
    "identifier": "FBBDDestination001",
    "name": "设置/目的地",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539814108320718850",
    "identifier": "FBAOrder0018",
    "name": "FBA物流/FBA订单/已退件",
  },
  {
    "businessOwnership": "kygj",
    "id": "1428258231742500866",
    "identifier": "SiteManagement004",
    "name": "客户端/公众号",
  },
  {
    "businessOwnership": "kygj",
    "id": "1440551481731846145",
    "identifier": "LandTransport001",
    "name": "头程/陆运头程",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539813103122210818",
    "identifier": "FBAOrder0012",
    "name": "FBA物流/FBA订单/已入库",
  },
  {
    "businessOwnership": "kygj",
    "id": "1437969073073426433",
    "identifier": "Workbench001",
    "name": "工作台",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412348546715649",
    "identifier": "pageView01",
    "name": "客户/客户管理/页面查看",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091067940212737",
    "identifier": "ReceivableBill001",
    "name": "收入/应收账单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1526144196423409666",
    "identifier": "FBAOrder001",
    "name": "FBA物流/FBA订单",
  },
  {
    "businessOwnership": "kygj",
    "id": "1539868602089824257",
    "identifier": "FBAOrder0019",
    "name": "FBA物流/FBA订单/全部",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417410333867315202",
    "identifier": "FBBDRemoteRule001",
    "name": "渠道/偏远查询",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417412768019058690",
    "identifier": "StaffMangement001",
    "name": "系统/员工管理",
  },
  {
    "businessOwnership": "kygj",
    "id": "1417403832310239233",
    "identifier": "FBOrder0024",
    "name": "专线物流/专线订单/已装箱",
  },
  {
    "businessOwnership": "kygj",
    "id": "1455439228737396737",
    "identifier": "FBOrder0027",
    "name": "专线物流/专线订单/已签收",
  },
  {
    "businessOwnership": "kygj",
    "id": "1468091147241918465",
    "identifier": "PayableBill001",
    "name": "成本/应付账单",
  },
]
`;
